### PR TITLE
#23626 : Editor settings color picker closes itself prematurely

### PR DESCRIPTION
**Pull request by Amssat221**
Fixed how the color picker would crash and close itself when browsing for a color.

### CODE REVIEW

**Comment by akien-mga** [comment]
You need to rebase your branch on top of the upstream master branch to fix merge conflicts and remove the unnecessary reverts. See [PR workflow](http://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#mastering-the-pr-workflow-the-rebase).

While doing so, you should also amend the commit log to explicitly state what the commit does, and not just "Update $file". See https://github.com/godotengine/godot/blob/master/CONTRIBUTING.md#contributing-pull-requests

File: editor/editor_settings.cpp [hunk 1]
```diff
@@ -330,9 +330,10 @@ void EditorSettings::_load_defaults(Ref<ConfigFile> p_extra_config) {
 	hints["interface/theme/preset"] = PropertyInfo(Variant::STRING, "interface/theme/preset", PROPERTY_HINT_ENUM, "Default,Alien,Arc,Godot 2,Grey,Light,Solarized (Dark),Solarized (Light),Custom", PROPERTY_USAGE_DEFAULT | PROPERTY_USAGE_RESTART_IF_CHANGED);
 	_initial_set("interface/theme/icon_and_font_color", 0);
 	hints["interface/theme/icon_and_font_color"] = PropertyInfo(Variant::INT, "interface/theme/icon_and_font_color", PROPERTY_HINT_ENUM, "Auto,Dark,Light", PROPERTY_USAGE_DEFAULT | PROPERTY_USAGE_RESTART_IF_CHANGED);
-	_initial_set("interface/theme/base_color", Color::html("#323b4f"));
-	hints["interface/theme/accent_color"] = PropertyInfo(Variant::COLOR, "interface/theme/accent_color", PROPERTY_HINT_NONE, "", PROPERTY_USAGE_DEFAULT | PROPERTY_USAGE_RESTART_IF_CHANGED);
+	//_initial_set("interface/theme/base_color", Color::html("#323b4f"));
 	_initial_set("interface/theme/accent_color", Color::html("#699ce8"));
+	hints["interface/theme/accent_color"] = PropertyInfo(Variant::COLOR, "interface/theme/accent_color", PROPERTY_HINT_NONE, "", PROPERTY_USAGE_DEFAULT | PROPERTY_USAGE_RESTART_IF_CHANGED);
+	_initial_set("interface/theme/base_color", Color::html("#323b4f"));
```
Comment (akien-mga) [inline editor/editor_settings.cpp]: Please keep `base_color` first, but move its `hints` before the `accent_color` definition.

**Comment by akien-mga** [comment]
Superseded by #27130.

**Comment by Amssat221** [comment]
Hey, the issue is deeper actually as it affects almost every parts of the editor settings. It seems like the problem is with the timer. It is set to 1.5 s and don't give users enough time to pick their colors. If the time runs out, the color picker close itself and apply the changes. If the user try to reopen the color picker  again it will crash and close immediately. Do we need to open another issue for it?  I removed the timer and everything seems to be working fine now. Is the timer really necessary? If it's not, I can resubmit a pull request.

**Comment by akien-mga** [comment]
It would be good to open an issue about it yeah. And you can also make a PR to propose the timer's removal, but I assume that it was added for a specific purpose, so the proper fix might be more involved (still, a PR is fine to kickstart discussions).

**Comment by Amssat221** [comment]
I found another way to solve the issue without removing the timer. I will
submit a pull request for it.

**Pull request closed without merging**